Added new DSC resource xVirtualMemory

nyanhp opened the pull request:
Added a new DSC resource xVirtualMemory to be able to set the paging file on a machine. Fixes #63 . In addition to the new resource, unit and integration tests and a sample script were provided as well.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/powershell/xcomputermanagement/64)
<!-- Reviewable:end -->

msftclas commented:
@__nyanhp__,
Thanks for your contribution.
To ensure that the project team has proper rights to use your work, please complete the Contribution License Agreement at https://cla.microsoft.com.

It will cover your contributions to all Microsoft-managed open source projects.
_Thanks,
Microsoft Pull Request Bot_

PlagueHO commented:
Really great job @nyanhp ! Thank you for submitting this! Sorry about all the nitpicky style changes - hope it's not too annoying!

---

Reviewed 4 of 7 files at r1, 4 of 4 files at r2.
Review status: all files reviewed at latest revision, 61 unresolved discussions.

---

*[README.md, line 81 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7bW7EOvCcbiy0ssiT:-Ke7bW7Fn-S6eFoIDN5H:b1ny5nk) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/README.md#L81)):*
> ```Markdown
>  
> ## xVirtualMemory
> xVirtualMemory resource is used to set the properties of the paging file on the local computer.
> ```

Please add CRLF above this line to meet Markdown rules.

---

*[README.md, line 83 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7beIKsn6gyB8LF06v:-Ke7beIKsn6gyB8LF06w:b1ny5nk) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/README.md#L83)):*
> ```Markdown
> xVirtualMemory resource is used to set the properties of the paging file on the local computer.
> xVirtualMemory has the following properties:
> * Type: The type of the paging settings, mandatory, out of "AutoManagePagingFile","CustomSize","SystemManagedSize","NoPagingFile"
> ```

Please add CRLF above this line to meet Markdown rules.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 16 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7bifUTYUZ5jqW6K4r:-Ke7bifVZ_36xhhs8AQC:b-z99m1g) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L16)):*
> ```PowerShell
>     )
> 
>     Write-Verbose 'Getting current page file settings'
> ```

Can you please convert positional parameter to named parameter?

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 27 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7bqpXPny3jHY1vmgW:-Ke7bqpXPny3jHY1vmgX:b1anecv) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L27)):*
> ```PowerShell
>     [bool] $isSystemManaged = (Get-CimInstance -ClassName Win32_ComputerSystem).AutomaticManagedPagefile
>     
>     if ($isSystemManaged) {
> ```

Move { to next line please.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 34 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7bvD1OFtIdmQ0S2wo:-Ke7bvD1OFtIdmQ0S2wp:b-z99m1g) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L34)):*
> ```PowerShell
>     $driveItem = [System.IO.DriveInfo] $Drive
> 
>     Write-Verbose "Pagefile was not automatically managed. Retrieving detailed page file settings with query Select * from Win32_PageFileSetting where SettingID='pagefile.sys @ $($driveItem.Name.Substring(0,2))'"
> ```

Can you please convert positional parameter to named parameter?

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 37 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7bzs8vdvS98OxQskX:-Ke7bzs8vdvS98OxQskY:b7i6vx4) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L37)):*
> ```PowerShell
> 
>     # Find existing page file settings by drive letter
>     $virtualMemoryInstance =  Get-CimInstance -Namespace root\cimv2 -Query "Select * from Win32_PageFileSetting where SettingID='pagefile.sys @ $($driveItem.Name.Substring(0,2))'"
> ```

Remove extra space after =

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 39 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7cojKZmxH1SdII4VG:-Ke7cojKZmxH1SdII4VH:bqxudxr) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L39)):*
> ```PowerShell
>     $virtualMemoryInstance =  Get-CimInstance -Namespace root\cimv2 -Query "Select * from Win32_PageFileSetting where SettingID='pagefile.sys @ $($driveItem.Name.Substring(0,2))'"
>     
>     if (-not $virtualMemoryInstance) {
> ```

Please move { to next line.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 44 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7csSOQltA1TAbZZzL:-Ke7csSOQltA1TAbZZzM:bqxudxr) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L44)):*
> ```PowerShell
>     }
> 
>     if ($virtualMemoryInstance.InitialSize -eq 0 -and $virtualMemoryInstance.MaximumSize -eq 0) {
> ```

Please move { to next line.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 47 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7cwBzOcIuMo4G5GnG:-Ke7cwBzOcIuMo4G5GnH:bqxudxr) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L47)):*
> ```PowerShell
>         $returnValue.Type = 'SystemManagedSize'
>     }
>     else {
> ```

Please move { to next line.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 83 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7dA2XdwYNCF-q-JMF:-Ke7dA2XdwYNCF-q-JMG:bqxudxr) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L83)):*
> ```PowerShell
>     $SystemInfo = Get-CimInstance -Class Win32_ComputerSystem
> 
>     switch($Type) {
> ```

Please move { to next line.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 84 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7dAiHo3t9DcXw7Z6Y:-Ke7dAiIOIVwGEckodEN:bqxudxr) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L84)):*
> ```PowerShell
> 
>     switch($Type) {
>         "AutoManagePagingFile" {
> ```

Please move { to next line.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 91 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7doi6ViqmY4KK1VKQ:-Ke7doi6ViqmY4KK1VKR:b-z99m1g) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L91)):*
> ```PowerShell
>             } 
> 
>             Write-Verbose 'Enabling AutoManagePagingFile'
> ```

Can you please convert positional parameter to named parameter?

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 97 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7dCVLzO1wZzkaC3t4:-Ke7dCVLzO1wZzkaC3t5:bqxudxr) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L97)):*
> ```PowerShell
>             break
>         }
>         "CustomSize" {
> ```

Please move { to next line.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 98 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7dDHkQnDStMIzcCl2:-Ke7dDHkQnDStMIzcCl3:bqxudxr) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L98)):*
> ```PowerShell
>         }
>         "CustomSize" {
>             if ($SystemInfo.AutomaticManagedPageFile) {
> ```

Please move { to next line.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 107 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7drUxThML85q5TpSf:-Ke7drUxThML85q5TpSg:b-z99m1g) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L107)):*
> ```PowerShell
>                 } 
> 
>                 Write-Verbose 'Disabling AutoManagePagingFile'
> ```

Can you please convert positional parameter to named parameter?

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 113 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7dFDUjiuHdPo4ADIW:-Ke7dFDUjiuHdPo4ADIX:bqxudxr) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L113)):*
> ```PowerShell
> 
>             $driveInfo = [System.IO.DriveInfo] $Drive
>             if (-not $driveInfo.IsReady) {
> ```

Please move { to next line.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 119 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7dwH78znBR3Z5dD8e:-Ke7dwH78znBR3Z5dD8f:b-z99m1g) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L119)):*
> ```PowerShell
>             $pageFileName = Join-Path $driveInfo.Name 'pagefile.sys'
> 
>             Write-Verbose ('Checking if a paging file already exists at {0}' -f $pageFileName)
> ```

Can you please convert positional parameter to named parameter?

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 121 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7dGW0nJEQ_4aQvkYM:-Ke7dGW0nJEQ_4aQvkYN:bqxudxr) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L121)):*
> ```PowerShell
>             Write-Verbose ('Checking if a paging file already exists at {0}' -f $pageFileName)
>             $existingPageFileSetting = Get-CimInstance -Namespace root\cimv2 -Query "Select * from Win32_PageFileSetting where SettingID='pagefile.sys @ $($driveInfo.Name.Substring(0,2))'"
>             if (-not $existingPageFileSetting) {
> ```

Please move { to next line.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 126 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7e0QoBgCf_XTUyKiD:-Ke7e0Qps-YpiEX3bukD:bqha4sw) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L126)):*
> ```PowerShell
> 
>             <# 
>             # New-CimInstance does not support properties InitialSize and MaximumSize. Therefore, create
> ```

Nitpick - no need for #
And on next line

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 138 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7e7eZKMjVcMd5u-fb:-Ke7e7eZKMjVcMd5u-fc:b-z99m1g) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L138)):*
> ```PowerShell
>             } 
> 
>             Write-Verbose ("Setting page file to {0}. Initial size {1}MB, maximum size {2}MB" -f $pageFileName, $InitialSize, $MaximumSize)
> ```

Can you please convert positional parameter to named parameter?

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 144 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7dIN14KFvIeKSEmW4:-Ke7dIN14KFvIeKSEmW5:bqxudxr) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L144)):*
> ```PowerShell
>             break
>         }
>         "SystemManagedSize" {
> ```

Please move { to next line.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 145 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7dJ25freGFxQosOxo:-Ke7dJ25freGFxQosOxp:bqxudxr) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L145)):*
> ```PowerShell
>         }
>         "SystemManagedSize" {
>             if ($SystemInfo.AutomaticManagedPageFile) {
> ```

Please move { to next line.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 152 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7e9WPkDxAi67VC-hY:-Ke7e9WQuP_WQo5tivzv:b-z99m1g) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L152)):*
> ```PowerShell
>                 } 
> 
>                 Write-Verbose 'Disabling AutoManagePagingFile'
> ```

Can you please convert positional parameter to named parameter?

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 158 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7dK07oTM9vjZ4SQE9:-Ke7dK07oTM9vjZ4SQEA:bqxudxr) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L158)):*
> ```PowerShell
> 
>             $driveInfo = [System.IO.DriveInfo] $Drive
>             if (-not $driveInfo.IsReady) {
> ```

Please move { to next line.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 162 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7eAfxJBvnLgaBn88r:-Ke7eAfy-kAqm_EbS3T8:b-gegoo0) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L162)):*
> ```PowerShell
>             }
> 
>             $pageFileName = Join-Path $driveInfo.Name 'pagefile.sys'
> ```

Can you please convert positional parameters to named parameters?

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 167 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7dLjaQsfZbI5KObYL:-Ke7dLjaQsfZbI5KObYM:bqxudxr) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L167)):*
> ```PowerShell
> 
>             $existingPageFileSetting = Get-CimInstance -Namespace root\cimv2 -Query "Select * from Win32_PageFileSetting where SettingID='pagefile.sys @ $($driveInfo.Name.Substring(0,2))'"
>             if (-not $existingPageFileSetting) {
> ```

Please move { to next line.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 181 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7eEIRiqO6v8QXclMf:-Ke7eEISy3FViNmy-uu6:b-z99m1g) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L181)):*
> ```PowerShell
>             } 
> 
>             Write-Verbose "Enabling system-managed page file on $pageFileName"
> ```

Can you please convert positional parameter to named parameter?

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 187 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7dMhk2GfAVTvMWzQk:-Ke7dMhk2GfAVTvMWzQl:bqxudxr) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L187)):*
> ```PowerShell
>             break
>         }
>         "NoPagingFile" {
> ```

Please move { to next line.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 188 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7dNd173CjIm8xqM4o:-Ke7dNd173CjIm8xqM4p:bqxudxr) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L188)):*
> ```PowerShell
>         }
>         "NoPagingFile" {
>             if ($SystemInfo.AutomaticManagedPageFile) {
> ```

Please move { to next line.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 199 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7dOUvJPLgbm2vFwFD:-Ke7dOUvJPLgbm2vFwFE:bqxudxr) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L199)):*
> ```PowerShell
> 
>             $driveInfo = [System.IO.DriveInfo] $Drive
>             if (-not $driveInfo.IsReady) {
> ```

Please move { to next line.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 206 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7dPQZw-9YzKZidBmc:-Ke7dPQZw-9YzKZidBmd:bqxudxr) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L206)):*
> ```PowerShell
>                 
>             $existingPageFileSetting = Get-CimInstance -Namespace root\cimv2 -Query "Select * from Win32_PageFileSetting where SettingID='pagefile.sys @ $($driveInfo.Name.Substring(0,2))'"
>             if ($existingPageFileSetting) {
> ```

Please move { to next line.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 207 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7eHpWUHAJQckKB2n2:-Ke7eHpWUHAJQckKB2n3:b-z99m1g) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L207)):*
> ```PowerShell
>             $existingPageFileSetting = Get-CimInstance -Namespace root\cimv2 -Query "Select * from Win32_PageFileSetting where SettingID='pagefile.sys @ $($driveInfo.Name.Substring(0,2))'"
>             if ($existingPageFileSetting) {
>                 Write-Verbose "Removing existing page file $($existingPageFileSetting.Name)"
> ```

Can you please convert positional parameter to named parameter?

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 216 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7dQWZ8pbgebqk92aF:-Ke7dQWZ8pbgebqk92aG:bqxudxr) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L216)):*
> ```PowerShell
>             break
>         }
>         default {
> ```

Please move { to next line.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 248 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7dSZMLI_hgvWkrl-O:-Ke7dSZMLI_hgvWkrl-P:bqxudxr) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L248)):*
> ```PowerShell
> 
>     switch($Type) {
>         "AutoManagePagingFile" {
> ```

Please move { to next line.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 252 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7dVx1lf3W58tfHna-:-Ke7dVx2UiisjbXn76jn:bqxudxr) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L252)):*
> ```PowerShell
>             break
>         }
>         "CustomSize" {
> ```

Please move { to next line.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 253 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7dUGoGqoeJbkJT6HQ:-Ke7dUGoGqoeJbkJT6HR:bqxudxr) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L253)):*
> ```PowerShell
>         }
>         "CustomSize" {
>             if ($SystemInfo.AutomaticManagedPageFile) {
> ```

Please move { to next line.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 260 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7dX4mxEti1QL7lx6X:-Ke7dX4mxEti1QL7lx6Y:bqxudxr) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L260)):*
> ```PowerShell
>             $driveInfo = [System.IO.DriveInfo] $Drive
>             $PageFile = Get-CimInstance -Class Win32_PageFileSetting -Filter "SettingID='pagefile.sys @ $($driveInfo.Name.Substring(0,2))'"
>             if (-not $PageFile) {
> ```

Please move { to next line.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 265 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7dYAzY4X9iYBQavv7:-Ke7dYB0AxeCS0hfMudw:bqxudxr) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L265)):*
> ```PowerShell
>             }
> 
>             if (-not ($PageFile.InitialSize -eq $InitialSize -and $PageFile.MaximumSize -eq $MaximumSize)) {
> ```

Please move { to next line.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 273 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7dZ2a3pnCKjobnNP9:-Ke7dZ2a3pnCKjobnNPA:bqxudxr) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L273)):*
> ```PowerShell
>             break
>         }
>         "SystemManagedSize" {
> ```

Please move { to next line.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 274 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7dZgyCswC85SA9sBa:-Ke7dZgyCswC85SA9sBb:bqxudxr) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L274)):*
> ```PowerShell
>         }
>         "SystemManagedSize" {
>             if ($SystemInfo.AutomaticManagedPageFile) {
> ```

Please move { to next line.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 281 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7d_gbMo2pmPQCe3gb:-Ke7d_gbMo2pmPQCe3gc:bqxudxr) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L281)):*
> ```PowerShell
>             $driveInfo = [System.IO.DriveInfo] $Drive
>             $PageFile = Get-CimInstance -Class Win32_PageFileSetting -Filter "SettingID='pagefile.sys @ $($driveInfo.Name.Substring(0,2))'"
>             if (-not $PageFile) {
> ```

Please move { to next line.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 286 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7daSb6g3MKtz7rVYK:-Ke7daSb6g3MKtz7rVYL:bqxudxr) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L286)):*
> ```PowerShell
>             }
> 
>             if (-not ($PageFile.InitialSize -eq 0 -and $PageFile.MaximumSize -eq 0)) {
> ```

Please move { to next line.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 294 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7dbXs8_RxhXW6LRzK:-Ke7dbXs8_RxhXW6LRzL:bqxudxr) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L294)):*
> ```PowerShell
>             break
>         }
>         "NoPagingFile" {
> ```

Please move { to next line.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 295 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7dct0VadzDK3beOIC:-Ke7dct157wRhxtxZ-RM:bqxudxr) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L295)):*
> ```PowerShell
>         }
>         "NoPagingFile" {
>             if ($SystemInfo.AutomaticManagedPageFile) {
> ```

Please move { to next line.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 303 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7de-P1ROV_8VVedt2:-Ke7de-P1ROV_8VVedt3:bqxudxr) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L303)):*
> ```PowerShell
>             $PageFile = Get-CimInstance -Class Win32_PageFileSetting -Filter "SettingID='pagefile.sys @ $($driveInfo.Name.Substring(0,2))'"
>                 
>             if ($PageFile) {
> ```

Please move { to next line.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 311 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7dfCzQGP9UcBDe_ZV:-Ke7dfCzQGP9UcBDe_ZW:bqxudxr) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L311)):*
> ```PowerShell
>             break
>         }
>         default {
> ```

Please move { to next line.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.schema.mof, line 5 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7edn6-cdPCGDQxwvo:-Ke7edn6-cdPCGDQxwvp:b-d32og2) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.schema.mof#L5)):*
> ```
> class MSFT_xVirtualMemory : OMI_BaseResource
> {
>     [Key] String Drive;
> ```

Can you please add description values to each field? This will help when we move over to auto-documentation generation from the MOF files (being worked on at the moment).

---

*[Tests/Integration/MSFT_xVirtualMemory.Integration.Tests.ps1, line 13 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7faKh8MsDczSFsTSq:-Ke7faKh8MsDczSFsTSr:beopacv) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/Tests/Integration/MSFT_xVirtualMemory.Integration.Tests.ps1#L13)):*
> ```PowerShell
>    Code in HEADER, FOOTER and DEFAULT TEST regions are standard and may be moved into
>    DSCResource.Tools in Future and therefore should not be altered if possible.
> #>
> ```

Can you please delete the header comment?

---

*[Tests/Integration/MSFT_xVirtualMemory.Integration.Tests.ps1, line 46 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7f3faneRXs1GMy2b7:-Ke7f3faneRXs1GMy2b8:b-gegoo0) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/Tests/Integration/MSFT_xVirtualMemory.Integration.Tests.ps1#L46)):*
> ```PowerShell
>         Context "Set page file to automatically managed" {
>             $CurrentConfig = "setToAuto"
>             $ConfigDir = (Join-Path $TestDrive $CurrentConfig)
> ```

Can you please convert positional parameters to named parameters?

---

*[Tests/Integration/MSFT_xVirtualMemory.Integration.Tests.ps1, line 47 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7f6nEOV3xpfuLnhfL:-Ke7f6nEOV3xpfuLnhfM:b-gegoo0) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/Tests/Integration/MSFT_xVirtualMemory.Integration.Tests.ps1#L47)):*
> ```PowerShell
>             $CurrentConfig = "setToAuto"
>             $ConfigDir = (Join-Path $TestDrive $CurrentConfig)
>             $ConfigMof = (Join-Path $ConfigDir "localhost.mof")
> ```

Can you please convert positional parameters to named parameters?

---

*[Tests/Integration/MSFT_xVirtualMemory.Integration.Tests.ps1, line 68 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7fOqMAHZUvxXmJx5p:-Ke7fOqNcRsofWcWx61k:b-gegoo0) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/Tests/Integration/MSFT_xVirtualMemory.Integration.Tests.ps1#L68)):*
> ```PowerShell
>         Context "Set page file to custom size" {
>             $CurrentConfig = "setToCustom"
>             $ConfigDir = (Join-Path $TestDrive $CurrentConfig)
> ```

Can you please convert positional parameters to named parameters?

---

*[Tests/Integration/MSFT_xVirtualMemory.Integration.Tests.ps1, line 69 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7fPRZUCw0nwj68Mh6:-Ke7fPRZUCw0nwj68Mh7:b-gegoo0) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/Tests/Integration/MSFT_xVirtualMemory.Integration.Tests.ps1#L69)):*
> ```PowerShell
>             $CurrentConfig = "setToCustom"
>             $ConfigDir = (Join-Path $TestDrive $CurrentConfig)
>             $ConfigMof = (Join-Path $ConfigDir "localhost.mof")
> ```

Can you please convert positional parameters to named parameters?

---

*[Tests/Integration/MSFT_xVirtualMemory.Integration.Tests.ps1, line 90 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7fRi73_EXoElHYdkx:-Ke7fRi73_EXoElHYdky:b-gegoo0) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/Tests/Integration/MSFT_xVirtualMemory.Integration.Tests.ps1#L90)):*
> ```PowerShell
>         Context "Set page file to system managed" {
>             $CurrentConfig = "setToSystemManaged"
>             $ConfigDir = (Join-Path $TestDrive $CurrentConfig)
> ```

Can you please convert positional parameters to named parameters?

---

*[Tests/Integration/MSFT_xVirtualMemory.Integration.Tests.ps1, line 91 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7fS9kiJCmyi3l3Tec:-Ke7fS9kiJCmyi3l3Ted:b-gegoo0) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/Tests/Integration/MSFT_xVirtualMemory.Integration.Tests.ps1#L91)):*
> ```PowerShell
>             $CurrentConfig = "setToSystemManaged"
>             $ConfigDir = (Join-Path $TestDrive $CurrentConfig)
>             $ConfigMof = (Join-Path $ConfigDir "localhost.mof")
> ```

Can you please convert positional parameters to named parameters?

---

*[Tests/Integration/MSFT_xVirtualMemory.Integration.Tests.ps1, line 112 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7fTe56CRGuabmi93J:-Ke7fTe56CRGuabmi93K:b-gegoo0) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/Tests/Integration/MSFT_xVirtualMemory.Integration.Tests.ps1#L112)):*
> ```PowerShell
>         Context "Set page file to none" {
>             $CurrentConfig = "setToNone"
>             $ConfigDir = (Join-Path $TestDrive $CurrentConfig)
> ```

Can you please convert positional parameters to named parameters?

---

*[Tests/Integration/MSFT_xVirtualMemory.Integration.Tests.ps1, line 113 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7fU_cPweOBLOqFPNa:-Ke7fU_cPweOBLOqFPNb:b-gegoo0) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/Tests/Integration/MSFT_xVirtualMemory.Integration.Tests.ps1#L113)):*
> ```PowerShell
>             $CurrentConfig = "setToNone"
>             $ConfigDir = (Join-Path $TestDrive $CurrentConfig)
>             $ConfigMof = (Join-Path $ConfigDir "localhost.mof")
> ```

Can you please convert positional parameters to named parameters?

---

*[Tests/Integration/MSFT_xVirtualMemory.Integration.Tests.ps1, line 141 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7fWVpoxoWdmUBUWdd:-Ke7fWVpoxoWdmUBUWde:bqs34co) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/Tests/Integration/MSFT_xVirtualMemory.Integration.Tests.ps1#L141)):*
> ```PowerShell
>     #endregion
> 
>     # TODO: Other Optional Cleanup Code Goes Here...
> ```

Can you delete this line?

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 133 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7fp_6r4WTxD1Ebatn:-Ke7fp_6r4WTxD1Ebato:ba3jstb) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/Tests/Unit/MSFT_xComputer.Tests.ps1#L133)):*
> ```PowerShell
>                     $Result = Get-TargetResource -Name $env:COMPUTERNAME
>                     $Result.GetType().Fullname | Should Be 'System.Collections.Hashtable'
>                     $Result.Keys | Sort-Object | Should Be @('Credential', 'CurrentOU', 'DomainName', 'JoinOU', 'Name',  'UnjoinCredential', 'WorkGroupName')
> ```

Is this an intentional fix to the tests? If so, no worries - I can see it is correct :grin:

---

*[Tests/Unit/MSFT_xVirtualMemory.Tests.ps1, line 46 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7hAppt0tqPTJC8fdX:-Ke7hAppt0tqPTJC8fdY:bvoqjvk) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/Tests/Unit/MSFT_xVirtualMemory.Tests.ps1#L46)):*
> ```PowerShell
>                 BeforeEach {
>                     Mock -CommandName Get-CimInstance -MockWith {
>                         return New-Object Object | 
> ```

Nitpick: Why not:
```powershell
[PSObject] @{ AutomaticManagegPageFile = $false; Name = 'D:\pagefile.sys' }
```
Also, if the Mock return is going to be used more than once (as is the case further down), you can store the object in a variable and return that from the mock to reduce code.

---

*[Tests/Unit/MSFT_xVirtualMemory.Tests.ps1, line 125 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7gaExoDONzXI16DLt:-Ke7gaEyRBfQnDNHv9dA:bcodqh6) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/Tests/Unit/MSFT_xVirtualMemory.Tests.ps1#L125)):*
> ```PowerShell
>                     MaximumSize = 1337
>                 } 
>                 It 'Should throw if no valid drive letter has been used' { { Set-TargetResource @testParameters } | Should Throw
> ```

Nitpick: Move inner block to it's own line to improve readability.

---

*[Tests/Unit/MSFT_xVirtualMemory.Tests.ps1, line 134 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7girDAeFKQM9i-iuC:-Ke7girDAeFKQM9i-iuD:bcodqh6) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/Tests/Unit/MSFT_xVirtualMemory.Tests.ps1#L134)):*
> ```PowerShell
>                     MaximumSize = 1337
>                 } 
>                 It 'Should throw if the drive is not ready' { { Set-TargetResource @testParameters } | Should Throw
> ```

Nitpick: Move inner block to it's own line to improve readability.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-:-Ke7hmf0c2SaOSpQcmSU:b-ruwdm4)*
<!-- Sent from Reviewable.io -->

nyanhp commented:
No worries @PlagueHO , a consistent style is very important! In Visual Studio with Style Cop it was easier though :) I fixed all styling issues and pushed my changes. Thank you for reviewing them!

Regarding the modified test for xComputer: Somehow when AppVeyor kicked in, the order of the results was checked very strictly. On my client, I had no problems with the test as it was before. Since my tests kept failing due to this, I changed the expected order and the order of the returned values to be alphabetic.

nyanhp commented:
Hi @PlagueHO , is there anything else I need to do? I fixed everything and pushed my code. Do I need to do things on Reviewable as well?

msftclas commented:
__@nyanhp__, thanks for signing the contribution license agreement. We will now validate the agreement and then the pull request.
<br />_Thanks, Microsoft Pull Request Bot_

PlagueHO commented:
Hi @nyanhp, no, you're good. Leave it with me and ill get a review done tonight after work! Really appreciate your work and patience! Sorry about the delay!

PlagueHO commented:
Hi @nyanhp - looking really good - just a few changes missed ( moving { to next line ). Will be good to go after those are fixed.

---

Reviewed 5 of 5 files at r3.
Review status: all files reviewed at latest revision, 14 unresolved discussions.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 248 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7dSZMLI_hgvWkrl-O:-KfogxBKKflb17jsibIf:bt0xsiy) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L248)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Please move { to next line.
</blockquote></details>

Missed this one.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 252 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7dVx1lf3W58tfHna-:-KfoiSDyLFlBZBkX1scW:bt0xsiy) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L252)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Please move { to next line.
</blockquote></details>

Missed this one.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 253 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7dUGoGqoeJbkJT6HQ:-Kfoh-O3FfTmHTu40zLY:bt0xsiy) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L253)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Please move { to next line.
</blockquote></details>

Missed this one.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 260 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7dX4mxEti1QL7lx6X:-Kfoh0YH8Q3IyKG0lKAM:bt0xsiy) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L260)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Please move { to next line.
</blockquote></details>

Missed this one.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 265 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7dYAzY4X9iYBQavv7:-Kfoh257PYNbZdIQaaxR:bt0xsiy) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L265)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Please move { to next line.
</blockquote></details>

Missed this one.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 273 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7dZ2a3pnCKjobnNP9:-Kfoh37BG50xzEB4M7hC:bt0xsiy) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L273)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Please move { to next line.
</blockquote></details>

Missed this one.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 274 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7dZgyCswC85SA9sBa:-Kfoh3aww-sJH0w2no0j:bt0xsiy) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L274)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Please move { to next line.
</blockquote></details>

Missed this one.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 281 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7d_gbMo2pmPQCe3gb:-Kfoh4HGfDaLHj6RD9LA:bt0xsiy) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L281)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Please move { to next line.
</blockquote></details>

Missed this one.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 286 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7daSb6g3MKtz7rVYK:-Kfoh58hq7IaxiOowtoR:bt0xsiy) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L286)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Please move { to next line.
</blockquote></details>

Missed this one.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 294 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7dbXs8_RxhXW6LRzK:-Kfoh5p5Y8Qgt_ZulDMt:bt0xsiy) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L294)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Please move { to next line.
</blockquote></details>

Missed this one.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 295 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7dct0VadzDK3beOIC:-Kfoh6x5ouENPgB9wMZA:bt0xsiy) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L295)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Please move { to next line.
</blockquote></details>

Missed this one.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 303 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7de-P1ROV_8VVedt2:-Kfoh7umsC_UCI8SF-Q0:bt0xsiy) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L303)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Please move { to next line.
</blockquote></details>

Missed this one.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 311 at r2](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-Ke7dfCzQGP9UcBDe_ZV:-KfohAq8Wa-_IWrbfXvi:bt0xsiy) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b4ed149e33b08ffec65f876352ae11137d5c4986/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L311)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Please move { to next line.
</blockquote></details>

Missed this one.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-:-KfoiZyIjUegMi1OSgj1:b-fxzswq)*
<!-- Sent from Reviewable.io -->

nyanhp commented:
Don't worry @PlagueHO . I was just asking since this was my first PR and I was not entirely sure if I had to do additional things. I think I formatted everything I missed before.

msftclas commented:
__@nyanhp__, thanks for signing the contribution license agreement. We will now validate the agreement and then the pull request.
<br />_Thanks, Microsoft Pull Request Bot_

PlagueHO commented:
Great job @nyanhp! Sorry, I missed a couple of issues (each function must have Comment Based help) though.

---

Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 1 at r4](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-KhpkrQwxeEVazYzhcSd:-KhpkrQxWMHItlONHQiU:b-fpb9p7) ([raw file](https://github.com/powershell/xcomputermanagement/blob/4ea3cb59d851954b71ed84aa6467320ef6f1aefc/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L1)):*
> ```PowerShell
> function Get-TargetResource
> ```

Doh! Something I nearly missed: All functions need to have Comment based help (see https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md#functions-have-comment-based-help)

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 63 at r4](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-KhpkyDgspB7U11d_BsA:-KhpkyDhKE2_IUEAGtd7:b-fpb9p7) ([raw file](https://github.com/powershell/xcomputermanagement/blob/4ea3cb59d851954b71ed84aa6467320ef6f1aefc/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L63)):*
> ```PowerShell
> }
> 
> function Set-TargetResource
> ```

Doh! Something I nearly missed: All functions need to have Comment based help (see https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md#functions-have-comment-based-help)

---

*[DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1, line 242 at r4](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-KhpkzOK1zRyMjPpGT-B:-KhpkzOK1zRyMjPpGT-C:b-fpb9p7) ([raw file](https://github.com/powershell/xcomputermanagement/blob/4ea3cb59d851954b71ed84aa6467320ef6f1aefc/DSCResources/MSFT_xVirtualMemory/MSFT_xVirtualMemory.psm1#L242)):*
> ```PowerShell
> }
> 
> function Test-TargetResource
> ```

Doh! Something I nearly missed: All functions need to have Comment based help (see https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md#functions-have-comment-based-help)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-:-KhpkSR_h6yDCRaN-gn1:b-hsck4o)*
<!-- Sent from Reviewable.io -->

nyanhp commented:
Hi @PlagueHO, done.
Out of curiosity: Why don't the other Get/Set/Test functions in the xComputerManagement resources have comment-based help? I can open a new issue for that and add the help.

PlagueHO commented:
Hi @nyanhp - 

Regards the comment based help:

over time the style guidelines have evolved somewhat - so earlier modules might not meet the current standards. This has introduced tech debt that we'd like to fix (but it takes time) - I'm slowly working my way through it (working on xCertificate ATM). We've also been trying to move the core resource kit modules over to meet HQRM standards (https://github.com/PowerShell/DscResources/blob/master/HighQualityModuleGuidelines.md). Comment Based help is one of the HQRM standards.

I will say that there has been a bit of too and fro over this particular one in the past and it was decided to keep it :grin:

Long story short: We'll want to add comment based help to any modules that are missing it.

I'll finish this review over the weekend (been a bit of a hectic week). Thanks for all your help and patience!

PlagueHO commented:
Actually, it wasn't a big review - so I did it now.

<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

---

Reviewed 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcomputermanagement/64#-:-KjHrRggrQVhtjDR6grj:b7701w4)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
@mbreakey3 - if you get a chance, could you merge this one for us? I think it's good to go. Cheers Dan
